### PR TITLE
Save state

### PR DESCRIPTION
**Pull request by ybaruti**

### CODE REVIEW

File: src/SpellingBee/saveState.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# State structure for puzzles
+
+import sqlite3
+import generateSubset
+
+class Puzzle:
+    
+    def __init__(self, keyLett, pangram):
+        self.keyLett = keyLett
+        self.pangram = pangram
```
Comment (JMLovegren) [inline src/SpellingBee/saveState.py:13]: Double checking what we're storing here. I'm pretty sure PANGRAM is refering to the unique set of 7 letters for the game. Right now, we're scoring based on pangram in the DB, and don't need to worry about the base starting word after those words are derived. I would change pangram variable name to something like uniqueLetters

File: src/SpellingBee/saveState.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        self.score = 0
+        self.maxScore = 0
+        self.foundWordList = []
+        self.rank = ' '
+        
+    def showKeyLetter(self):
+        return self.keyLett
+    
+    def showUniqueLetters(self):
```
Comment (JMLovegren) [inline src/SpellingBee/saveState.py:22]: to reiterate comment on pangram ambiguity, would be consistent with fucntion naming here

File: src/SpellingBee/saveState.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    # All words for current puzzle
+    def wordListStorage(self):
+       return generateSubset.getAllWordsFromPuzzle(self.pangram, self.keylett)
+    
+    
+    # Returns a number
+    def showMaxScore(self):
+        return self.maxScore
+
+    # Returns a string of all the words
```
[the remark this answers is not in the document]
Comment (JMLovegren) [reply src/SpellingBee/saveState.py:35]: I may have misunderstood. I'm pretty sure you *are* storing the found words as a list, this function is just joining them together. That makes sense

**Review by JMLovegren — CHANGES_REQUESTED** [review]
Commented code with input, handful of changes I'd like to see, but overall really good!

**Comment by ybaruti** [comment]
dope thank you i will review and make the changes

**Comment by JMLovegren** [comment]
@ybaruti if my comments didn't populate lmk

**Comment by ybaruti** [comment]
nope I got them

**Review by JMLovegren — APPROVED** [review]
Thank you for changing that. Looks like a good core for our system!

**Pull request merged**